File: frontend/ubre/src/app/dtos/report-summary-dto.ts
/**
 * Kumulativne i prosečne vrednosti za ceo opseg datuma izveštaja.
 */
export interface ReportSummaryDto {
  /** Ukupan broj vožnji u opsegu. */
  totalRides: number;
  /** Ukupno pređenih kilometara. */
  totalDistanceKm: number;
  /** Ukupno novca (negativno potrošeno, pozitivno zarađeno). */
  totalAmountMoney: number;
  /** Prosečan broj vožnji po danu. */
  averageRidesPerDay: number;
  /** Prosečna distanca po danu (km). */
  averageDistancePerDay: number;
  /** Prosečan novac po danu. */
  averageMoneyPerDay: number;
}

File: backend/src/main/java/com/ubre/backend/dto/ReportSummaryDto.java
package com.ubre.backend.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
public class ReportSummaryDto {
    private Integer totalRides;
    private Double totalDistanceKm;
    private Double totalAmountMoney;
    private Double averageRidesPerDay;
    private Double averageDistancePerDay;
    private Double averageMoneyPerDay;
}
